corehq/apps/app_manager/models.py

```diff
@@ -1519,18 +1520,48 @@ class MappingItem(DocumentSchema):
     value = DictProperty()
 
     @property
+    def treat_as_expression(self):
+        """
+        Returns if whether the key can be treated as a valid expression that can be included in
+        condition-predicate of an if-clause for e.g. if(<expression>, value, ...)
+        """
+        special_chars = '{}()[]=<>."\'/'
```

Comment:
Seems simpler to say what the legal characters are, since there are fewer of them - anything matches `re'\W'` is an expression, right?